### PR TITLE
Extract identity number from word list

### PR DESCRIPTION
**Pull request by nmattia**
This splits the identity number input from the grid of recovery words. Having the identity number as part of the grid was confusing for some users.

The treatment is applied on three different screens (all screens that show the word grid):

* When the recovery phrase is created, the identity is separate (though still copied to the clipboard)
* When confirming/verifying the recovery phrase, the identity is separate and uses colors for validity feedback (unlike words that also use an icon). We may revisit this for more clarity.
* When recovering using the phrase, the identity input is also split out.

The `anchorInput` component is adapted to accomodate e.g. more classes and more callbacks (the new screens now make use of `oninput`).

The `Chan` gets a new helper, `zip()`, to zip two `Chan`s together.

---

Recovery confirmation:

https://github.com/dfinity/internet-identity/assets/6930756/67f3e3fc-b42f-4893-b2f6-1ea6f0b45b45

### CODE REVIEW

File: src/frontend/src/styles/main.css [hunk 1]
```diff
@@ -110,6 +110,7 @@
   --vc-brand-orange: #f15a24;
   --vc-brand-yellow: #fbb03b;
   --vc-brand-pink: #ed1e79;
+  --vc-brand-red: red;
```
[the remark this answers is not in the document]
Comment (meodai) [reply src/frontend/src/styles/main.css:113]: I think I would still use `var(--rc-error)` I know its pink, but its the error color we use everywhere, if you dont wan't to I would still call it `--rc-error-strong: var(--vc-brand-red);` or something

File: src/frontend/src/styles/main.css [hunk 2]
```diff
@@ -2383,6 +2384,16 @@ a.c-action-list__item {
...
+  /* The halo around the input, used for style & feedback */
+  --halo: linear-gradient(270deg, var(--rg-brand));
+}
+
+.c-input--anchor__wrap--error {
+  --halo: var(--vc-brand-red);
+}
+
+.c-input--anchor__wrap--valid {
+  --halo: var(--vc-brand-green);
```
Comment (meodai) [inline src/frontend/src/styles/main.css:2396]: `var(--rc-positive)`

**Review by meodai — APPROVED** [review]
Beside the comments I made LGTM

File: src/frontend/src/styles/main.css (hunk 1 above)
[the remark this answers is not in the document]
Comment (meodai) [reply src/frontend/src/styles/main.css:113]: yeah maybe you can just call it --vc-red if you feel the usual error color does not cut it for you

File: src/frontend/src/styles/main.css [hunk 3]
```diff
@@ -105,11 +105,13 @@
   --vc-shadow: rgba(32, 33, 36, 0.6);
   --vc-night: rgba(32, 33, 36, 0.8);
 
+  --vc-red: #ff0000;
   --vc-silver: #d9d9d9;
 
   --vc-brand-orange: #f15a24;
   --vc-brand-yellow: #fbb03b;
   --vc-brand-pink: #ed1e79;
+  --vc-brand-red: red;
```
Comment (nmattia) [inline src/frontend/src/styles/main.css:114]:
```suggestion
```

File: src/frontend/src/flows/recovery/confirmSeedPhrase.ts [hunk 4]
```diff
@@ -50,17 +53,74 @@ const confirmSeedPhraseTemplate = ({
...
-  const disabled = wordsOk.map((ok) => !ok);
+  const disabled = identityInputState
+    .zip(wordsOk)
+    .map(([idOk, wdsOk]) => !(idOk === "correct" && wdsOk));
+
+  // The anchor input element/component
+  const anchorInput = mkAnchorInput({
+    onInput: (a) => {
+      // When inputting a value, we don't report incorrect values as the
+      // user type; we only react if the value is actually correct to let
```
Comment (nmattia) [inline src/frontend/src/flows/recovery/confirmSeedPhrase.ts:79]:
```suggestion
      // user types; we only react if the value is actually correct to let
```

File: src/frontend/src/components/anchorInput.ts [hunk 5]
```diff
@@ -104,7 +124,11 @@ export const mkAnchorInput = ({
         class="c-input c-input--vip c-input--centered c-input--spacious"
         placeholder="Internet Identity"
         value="${ifDefined(userNumber?.toString())}"
-        @input=${inputFilter(isDigits, onBadInput)}
```
[the remark this answers is not in the document]
Comment (nmattia) [reply src/frontend/src/components/anchorInput.ts:107]: https://github.com/dfinity/internet-identity/pull/1699

**Pull request merged**